### PR TITLE
feat: add createMtlsSession function

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+
+async function plugin (app, options) {
+  const clientsRole = options.mtlsClientsRole ?? null
+  const commonNameDomain = `.${options.mtlsCommonNameDomain}`
+
+  app.decorateRequest('createMtlsSession', async function () {
```
Comment (mcollina) [inline index.js:9]:
```suggestion
  app.decorateRequest('user')
  app.decorateRequest('createMtlsSession', async function () {
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const certificate = this.raw.socket.getPeerCertificate(false)
```
Comment (mcollina) [inline index.js:10]: You should throw a proper error in case this is not available (for example in case of `inject()`

**Review by mcollina — COMMENTED** [review]
How could we support `inject()`?

**Comment by ivan-tymoshenko** [comment]
> How could we support `inject()`?

You know more about undici and light-my-request. Can we add a `dispatcher` option to the `fastify.inject`options and make it work with undici dispatchers?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**